### PR TITLE
🔥 (#1565) Remove incentives vault from Compound

### LINKED ISSUE #1565: Remove `IncentivesVault`

### PR DESCRIPTION
**Pull request by MerlinEgalite**
# Pull Request

## Issue(s) fixed

This pull request fixes #1565

### CODE REVIEW

**Review by Rubilmax — APPROVED** [review]

**Review by pakim249CAL — APPROVED** [review]

File: src/compound/interfaces/IMorpho.sol [hunk 1]
```diff
@@ -35,7 +34,7 @@ interface IMorpho {
     function interestRatesManager() external view returns (IInterestRatesManager);
     function rewardsManager() external view returns (IRewardsManager);
     function positionsManager() external view returns (IPositionsManager);
-    function incentiveVault() external view returns (IIncentivesVault);
+    function incentiveVault() external view returns (address);
```
Comment (MathisGD) [inline src/compound/interfaces/IMorpho.sol:37]:
```suggestion
    function incentivesVault() external view returns (address);
```

File: src/compound/Morpho.sol [hunk 2]
```diff
@@ -146,9 +146,8 @@ contract Morpho is MorphoGovernance {
 
     /// @notice Claims rewards for the given assets.
     /// @param _cTokenAddresses The cToken addresses to claim rewards from.
-    /// @param _tradeForMorphoToken Whether or not to trade COMP tokens for MORPHO tokens.
```
Comment (MathisGD) [inline src/compound/Morpho.sol:149]: maybe explain that the variable is deprecated ?

**Review by MathisGD — CHANGES_REQUESTED** [review]

**Comment by MathisGD** [comment]
Why not also removing it from aave-v2 ?

File: src/compound/MorphoStorage.sol [hunk 3]
```diff
@@ -53,7 +52,7 @@ abstract contract MorphoStorage is OwnableUpgradeable, ReentrancyGuardUpgradeabl
     /// CONTRACTS AND ADDRESSES ///
 
     IPositionsManager public positionsManager;
-    IIncentivesVault public incentivesVault;
+    address public incentivesVault; // Deprecated.
```
Comment (MathisGD) [inline src/compound/MorphoStorage.sol:55]: I would also remove the getter I thing
```suggestion
    address internal incentivesVault; // Deprecated.
```

**Comment by MerlinEgalite** [comment]
> Why not also removing it from aave-v2 ?

You can check it's already removed

File: src/compound/MorphoStorage.sol (hunk 3 above)
Comment (MerlinEgalite) [reply src/compound/MorphoStorage.sol:55]: It would change the abi so I don't think we should do it

**Comment by MathisGD** [comment]
Oh ok, I was looking on main

**Review by MathisGD — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Pull request merged**